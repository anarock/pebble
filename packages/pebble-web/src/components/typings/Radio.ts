import { Omit } from "utility-types";
import { ControlProps } from "./Control";

export type RadioProps = Omit<ControlProps, "type" | "children">;
