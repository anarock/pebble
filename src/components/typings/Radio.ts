import { Omit } from "utility-types";
import { ControlProps } from "./Control";

export type RadioProps<OptionType> = Omit<ControlProps<OptionType>, "type">;
