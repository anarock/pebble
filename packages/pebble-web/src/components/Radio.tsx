import * as React from "react";
import { RadioProps } from "./typings/Radio";
import Control from "./shared/Control";

export default function Radio<OptionType>(props: RadioProps<OptionType>) {
  return <Control {...props} type="radio" />;
}
