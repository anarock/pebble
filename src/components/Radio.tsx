import * as React from "react";
import { RadioProps } from "./typings/Radio";
import Control from "./shared/Control";

const Radio: React.SFC<RadioProps> = props => (
  <Control {...props} type="radio" />
);
export default Radio;
