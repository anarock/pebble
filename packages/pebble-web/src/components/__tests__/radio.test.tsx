import * as React from "react";
import Radio from "../Radio";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import sinon from "sinon";

describe("Radio", () => {
  test("snapshot", () => {
    const radio = renderer.create(
      <Radio
        checked
        onChange={() => {}}
        value="radio"
        label="I am radio button"
      />
    );

    const tree = radio.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should call onChange on click with correct arguments", () => {
    const spy = sinon.spy();

    const radio = mount(
      <Radio onChange={spy} value="radio" label="I am radio button" />
    );

    radio.simulate("click");

    expect(
      spy.calledWith({
        value: "radio",
        checked: true
      })
    ).toBeTruthy();
  });

  test("should not call onChange if disabled", () => {
    const spy = sinon.spy();

    const radio = mount(
      <Radio onChange={spy} value="radio" label="I am radio button" disabled />
    );

    radio.simulate("click");

    expect(spy.called).toBeFalsy();
  });
});
