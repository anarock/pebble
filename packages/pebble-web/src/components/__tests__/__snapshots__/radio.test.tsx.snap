// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio snapshot 1`] = `
.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  padding: 10px 0;
  position: relative;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 400;
  color: #101721;
  font-size: 14px;
}

.emotion-1[data-disabled='true'] {
  cursor: not-allowed;
}

.emotion-0 {
  margin-right: 10px;
  font-size: 16px;
}

[data-disabled='true'] .emotion-0 {
  color: #E0E0E0;
}

<div
  aria-checked={true}
  className="emotion-1"
  onClick={[Function]}
  role="radio"
  tabIndex={0}
>
  <i
    className="pi pi-radio-selected emotion-0"
    style={
      Object {
        "color": "#6161FF",
        "paddingTop": 2,
      }
    }
  />
   
  I am radio button
</div>
`;
